Dev Branch

malivinayak opened the pull request.

malivinayak committed:
Update README.md

malivinayak committed:
Initial Project Idea and implantation approach

malivinayak committed:
Initial Project Idea and implantation approach

malivinayak committed:
Removing extra blank 9. Point

malivinayak committed:
Removing extra blank 9. Point

malivinayak committed:
Adding Table of Content

malivinayak committed:
Adding Table of Content

malivinayak committed:
Removing Replicated Content

malivinayak committed:
Removing Replicated Content

malivinayak committed:
Added new idea

malivinayak committed:
Added new idea

malivinayak committed:
Merge branch 'dev' of https://github.com/malivinayak/Taxi-Management-System into dev

malivinayak committed:
initial setup

malivinayak committed:
Update README.md

malivinayak committed:
Server Up successfully

malivinayak committed:
DB Plan with ERD

malivinayak committed:
Installed OracleDB

malivinayak committed:
API routes and registration ...

malivinayak committed:
Dev Commit

malivinayak committed:
SQL File

malivinayak committed:
Added connection

malivinayak committed:
Driver Registration

malivinayak committed:
Setup Login

OutdatedGuy reviewed (dismissed):
LGTM!

malivinayak committed:
DataBase

malivinayak committed:
UserRegistration API Route

malivinayak committed:
fs package added

malivinayak committed:
.gitignore

malivinayak committed:
Insertion Query Format

OutdatedGuy commented on src/backend/routes/registration/userRegistration.js:
```diff
@@ -1,9 +1,20 @@
+import fs from 'fs';
+
+let libPath;
+if (process.platform === 'win32') {           // Windows
+    libPath = 'C:\\oracle\\instantclient_19_12';
+} else if (process.platform === 'darwin') {   // macOS
+    libPath = process.env.HOME + '/Downloads/instantclient_19_8';
+}
```
Is there a need to use absolute path here?

malivinayak replied:
Yes
If we don't use this the it throws error

OutdatedGuy commented on src/package.json:
```diff
@@ -26,6 +26,7 @@
   "dependencies": {
     "dotenv": "^16.0.3",
     "express": "^4.18.2",
+    "fs": "^0.0.1-security",
```
"fs" is inbuilt library in nodejs. You don't have to install a dependency.

malivinayak committed:
remove unnessesary vriables

malivinayak committed:
Driver registartion unique checks

malivinayak committed:
userspace settings

OutdatedGuy committed:
feat: added user registration frontend ui code (#21)

* feat: added user registration fromtend ui code

* feat: added pattern check for input fields

OutdatedGuy committed:
feat: added driver registration frontend ui code (#24)

OutdatedGuy committed:
fix: typo for birth date (#23)

malivinayak committed:
Fix: user registration variable (#22)

* Parsing Variables to respective data types

* making userID same as userName

* Removing console log

* changed dbo to birthDate

OutdatedGuy committed:
feat: added home page ui (#28)

malivinayak committed:
API Route - Login (#31)

* API Route - Login

* typo fix

malivinayak committed:
Login UI (#33)

malivinayak committed:
API Route: Driver registration (#16)

* checking unique aadhar number and insert query

* update

* INsert query for driver registration

* Removing Unessesary comment

* resolve: AADHAR_NUMBER to uppercase

* Capitalize the DB Variables

malivinayak committed:
Added all API Routes (#36)

* Adding all API Routes

* fixing typo

* added new line at the end

* Added newline between different sections

malivinayak committed:
API Route: Credentials (#37)

* Oracledb libpath added

* validating token and returning userName

malivinayak committed:
API Route: session end (#39)

* Oracledb libpath added

* Update token to null after session end

malivinayak committed:
API Route: User Query (#43)

* Fix: Route import path

* initial setup

* fix: userQuery Route

* API Route : User Query

* getting user address and removing console.log

malivinayak committed:
Fix API Route : for Admin Query (#45)

* Adding all API Routes

* fixing typo

* added new line at the end

* Added newline between different sections

* API Route: Driver registration (#16)

* checking unique aadhar number and insert query

* update

* INsert query for driver registration

* Removing Unessesary comment

* resolve: AADHAR_NUMBER to uppercase

* Capitalize the DB Variables

* API Route: Credentials (#37)

* Oracledb libpath added

* validating token and returning userName

* API Route: session end (#39)

* Oracledb libpath added

* Update token to null after session end

* Fix: Admin Query import module path

malivinayak committed:
API Route: Driver Query (#47)

* adding query and response

* updating query as per  filer requirement

* Adding new line at the end

malivinayak committed:
Parsing variable to expected data-type (#50)

OutdatedGuy committed:
feat!: using session storage to store user token (#54)

- Much secure than having it in url
- Other cannot see the token
- Token cannot be changed in url

OutdatedGuy committed:
feat: create admin login page ui (#55)

malivinayak committed:
fix(api_route): typo at admin login query (#57)

OutdatedGuy committed:
feat: created Admin dashboard (#58)

* feat: admin dashboard ui with user query form

* feat: added driver query form to admin dashboard

* feat: added trip query form to admin dashboard

* feat: added custom query form to admin dashboard

OutdatedGuy committed:
fix: admin session token not expire on page reload (#65)

OutdatedGuy committed:
feat: user dashboard with all functionalities (#66)

1. Can book trip with filled details
2. Can retrieve trip history and view in table format
3. Can Delete all current trip history
4. Can get current wallet balance
5. Can add more amount to wallent

patilprthm committed:
Doc: uploaded initial PPT, SRS, Synopsis  files (#41)

* Create README.md

* Add files via upload

* Create README.md

* Create DeleteME.txt

* Delete DeleteME.txt

* Create README.md

* Add files via upload

* Create README.md

* Create DeleteME.txt

* Delete DeleteME.txt

* API Route - Login (#31)

* API Route - Login

* typo fix

* Login UI (#33)

* unnecessary folder

* SRS was uploaded in the folder appropriately

* PPT for Idea Presentation

* PDF version of PPT of Idea Presentation

* Create Synopsis

* Delete Synopsis

* Update README.md

* Add files via upload

* Update README.md

* Updated relative path instead of embedded link

* Updated SRS with it's relative path

* Updated SRS with it's relative path

* Create README.md

* Deleted README.md file 

Created while making SRS folder

* Created Synopsis folder

* Uploaded Synopsis file

* Updated synopsis in README.md file

* Updated Heading in README.md file

Co-authored-by: malivinayak <66154908+malivinayak@users.noreply.github.com>
Co-authored-by: Vinayak Mali <malivinayak012@gmail.com>
Co-authored-by: OutdatedGuy <everythingoutdated@gmail.com>

malivinayak committed:
Fix(api_route): login and token not clear issue (#68)

* fix: after session token not resting issue

* fix: Driver login issue

malivinayak committed:
fix: token is not resting to null issue (#71)

Co-authored-by: OutdatedGuy <everythingoutdated@gmail.com>

malivinayak committed:
API Route : get-balance (#74)

* adding getBalance route in index.js

* get Balance API Route

* fixing typo

* removing console log

* removing console log

malivinayak committed:
API Route: Trip Booking (#70)

* Trip Booking API Route

* updating CBS entries too

* removing console log

* removing unnecessary variable

* removing unnecessary variable

* setting booking rent to zero

* correcting error response

malivinayak committed:
API Route: add money (#75)

* fix: add-money cbs route

* API addMoney

* removing extra lines

* setting minimum deposit amount to 10

* setting maximum limit

* updating minimum amount to 100

* standardizing the error message

OutdatedGuy committed:
feat: trip rating tab for user dashboard (#80)

malivinayak committed:
API Route: withdraw money (#76)

* fix: withdraw-money cbs route

* withdraw money API

* setting withdrawal limit

* adding DB connection

* autocommit true

* removing extra lines

* setting min withdrawal to 100

* standardizing the error message

* acc minimum safety deposit case

OutdatedGuy committed:
feat: driver dashboard with all functionalities (#82)

1. Can get all available trips with details in table format
2. Can accept trip with trip id
3. Can retrieve trip history and view in table format
4. Can Delete all current trip history
5. Can get current wallet balance
6. Can withdraw money from wallet

malivinayak committed:
fix: driver token getting removed in frontend (#84)

OutdatedGuy committed:
fix: trip id format not correct (#87)

malivinayak committed:
getAvailable trip (except ac and vehicle type) (#89)

malivinayak committed:
API Route: trip history (#88)

* initial setup

* initial setup

* Query Logic

* initial setup

* Query Logic

* successful retrieval of user trip with status 1

* with status - 1

* successful retrieval  of user trip with status 0

* retrieval of driver trip info

* fix: user and driver history query

* beautifying the code

* initial setup

* Query Logic

* successful retrieval of user trip with status 1

* with status - 1

* successful retrieval  of user trip with status 0

* retrieval of driver trip info

* fix: user and driver history query

* beautifying the code

OutdatedGuy committed:
refactor: using currency symbol instead (#85)

Co-authored-by: Vinayak Mali <malivinayak012@gmail.com>

malivinayak committed:
API Route : Accept Trip (#97)

* creating tripAccept Route

* updating index.js file for route

* complete API route fro trip acceptance for driver

malivinayak committed:
API Route : delete trips (#91)

* setting up API route

* delete trip history route

* setting up API route

* delete trip history route

* updating query

* updating query to delete driver history

malivinayak committed:
updating fare rate as per user ac and car choice (#92)

malivinayak committed:
fix: retrived data for user not showing (#90)

malivinayak committed:
API Route: Rateable trip (#98)

* initial setup rateableTrip

* API rateable trip

malivinayak committed:
API Route: Admin Trip Query (#95)

* complete API Route for admin trip query

* adding new line at end

malivinayak committed:
API Route: Custom query (#61)

* initial setup for customQuery

* removing static fields

* removing extra whitespaces from end

* sending query error to alert box

* removing loop

* console log body

* removing whitespace and semicolon if any

* successful data retrieval as per custom query

* blocking insertion operation

* blocking update operation

* updating emoji

* restricting all access except select

* sending proper error of failure to correct query syntax

* replacing table name

* not sending password to retrivedData

* removing console.log

malivinayak committed:
fix: status bug (#100)

malivinayak committed:
API Route : Rate Trip (#99)

* setting up route

* initial setup

* sending temp response

* fixing rateable trip route

* API Route TRIP RATING

malivinayak committed:
fix: trip ID not showing after trip booking (#105)

malivinayak committed:
updating alert msg (#104)

malivinayak committed:
Fix replace end time to accept time (#103)

* trip history end time to accept time

* admintrip history end time to accept time

malivinayak committed:
Fix API: updating trip end time (#102)

* updating acceptance time

* removing unwanted file

malivinayak committed:
removing console log (#101)

OutdatedGuy committed:
feat: added range for start and end time trip query (#106)

malivinayak committed:
Updating Main `README.md` File (#107)

* Uploading UI screens

* moving UI to img directory

* uploading POSTER.PNG

* updating readme

* Functions And Features

* adding future scope and removing product feature

* uploading TEAM image

* updating UI screens in readme

* fixing poster url

* fixing intention

* updating index

* updating README

* Removing extra new lines

* Adding Advance Database Concepts

* fix: Reference URL
